src/main/java/seedu/address/model/person/person/Role.java

```diff
@@ -0,0 +1,64 @@
+package seedu.address.model.person.person;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+import java.util.Arrays;
+import java.util.stream.Collectors;
+
+/**
+ * Represents a Person's role in the address book.
+ * Guarantees: immutable;
+ */
+public class Role {
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "Role should only be a " + getMessageConstraints();
+    public static final String VALIDATION_REGEX = getValidationRegex();
+    public final RoleEnum value;
+
+    /**
+     * Constructs a {@code Role}.
+     *
+     * @param role A valid role.
+     */
+    public Role(String role) {
+        requireNonNull(role);
+        checkArgument(isValidRole(role), MESSAGE_CONSTRAINTS);
+        value = Arrays.stream(RoleEnum.values()).filter(x -> x.getArgument().equals((role))).findFirst().get();
+    }
+
+    private static final String getMessageConstraints() {
+        return Arrays.stream(RoleEnum.values()).map(role -> role.toString() + " (" + role.getArgument() + ")")
```

Comment:
I think maybe its better to switch the role.toString and role.getArgument() here as people may think role.toString is what they shld key in. Just a nitpick.